src/database.js

```diff
@@ -0,0 +1,166 @@
+const fs = require('fs');
+const {decode, encode} = require('./cipher');
+const path = require('path');
+
+// ID length in bytes
+const idSize = 2;
+
+const toBytes = (data) => {
+  const idBuffer = Buffer.alloc(idSize);
+  idBuffer.writeUInt16BE(data.lastIdx);
+  // Write json string with default UTF-8 encoding
+  const jsonBuffer = Buffer.from(JSON.stringify(data.entities));
+  return Buffer.concat([idBuffer, jsonBuffer]);
+};
+
+// Parses decoded database content into entities.
+const parse = (bytes) => {
+  if (bytes.length === 0) return {'entities': []};
+  /*
+   First two bytes represent ID counter, which is incremented on appending.
+   It is stored in big endian format as unsigned int16.
+   */
+  const lastIdx = bytes.readUInt16BE();
+  /*
+   All other information is just a UTF-8 string with JSON data.
+   Buffer.toString() uses UTF-8 encoding by default.
+   */
+  const info = bytes.subarray(idSize).toString();
+  const entities = JSON.parse(info);
+  if (!Array.isArray(entities)) {
+    throw new SyntaxError();
+  }
+  return {
+    lastIdx,
+    'entities': entities,
+  };
+};
+
+// Multiplying ID size in bytes by number of bits in a byte
+const idMax = 2 ** (idSize * 8) - 1;
+
+const isEntityCorrect = (entity) => {
+  for (const value of Object.values(entity)) {
+    if (typeof value === 'object') {
+      if (!isEntityCorrect(value)) return false;
+    } else if (typeof value !== 'string' && typeof value !== 'number') {
+      return false;
+    }
+  }
+  return true;
+};
+
+/** Implementation of single-file entity{key-value} storage */
+class SimpleDatabase {
+  /**
```

Comment:
I'd rather take this class out to a separate file.

Reply:
But there are lots of helper functions, I don't want to export them anywhere